azurerm/internal/services/apimanagement/api_management_api_release.go

```diff
@@ -0,0 +1,142 @@
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/pluginsdk"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func resourceApiManagementApiRelease() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Create: resourceApiManagementApiReleaseCreateUpdate,
+		Read:   resourceApiManagementApiReleaseRead,
+		Update: resourceApiManagementApiReleaseCreateUpdate,
+		Delete: resourceApiManagementApiReleaseDelete,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Create: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Read:   pluginsdk.DefaultTimeout(5 * time.Minute),
+			Update: pluginsdk.DefaultTimeout(30 * time.Minute),
+			Delete: pluginsdk.DefaultTimeout(30 * time.Minute),
+		},
+
+		Importer: pluginsdk.ImporterValidatingResourceId(func(id string) error {
+			_, err := parse.ApiReleaseID(id)
+			return err
+		}),
+
+		Schema: map[string]*pluginsdk.Schema{
+			"name": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.ApiManagementChildName,
+			},
+
+			"api_id": {
+				Type:         pluginsdk.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validate.ApiID,
+			},
+
+			"notes": {
+				Type:         pluginsdk.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringIsNotEmpty,
+			},
+		},
+	}
+}
+func resourceApiManagementApiReleaseCreateUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
+	subscriptionId := meta.(*clients.Client).Account.SubscriptionId
+	client := meta.(*clients.Client).ApiManagement.ApiReleasesClient
+	ctx, cancel := timeouts.ForCreate(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	apiId, err := parse.ApiID(d.Get("api_id").(string))
+	if err != nil {
+		return err
+	}
+	id := parse.NewApiReleaseID(subscriptionId, apiId.ResourceGroup, apiId.ServiceName, apiId.Name, name)
+
+	if d.IsNewResource() {
+		existing, err := client.Get(ctx, apiId.ResourceGroup, apiId.ServiceName, apiId.Name, name)
+		if err != nil {
+			if !utils.ResponseWasNotFound(existing.Response) {
+				return fmt.Errorf("checking for existing %q: %+v", id, err)
+			}
+		}
+		if !utils.ResponseWasNotFound(existing.Response) {
+			return tf.ImportAsExistsError("azurerm_api_management_api_release", id.ID())
+		}
+	}
+
+	parameters := apimanagement.APIReleaseContract{
+		APIReleaseContractProperties: &apimanagement.APIReleaseContractProperties{
+			APIID: utils.String(d.Get("api_id").(string)),
+			Notes: utils.String(d.Get("notes").(string)),
+		},
+	}
+
+	if _, err := client.CreateOrUpdate(ctx, apiId.ResourceGroup, apiId.ServiceName, apiId.Name, name, parameters, ""); err != nil {
```

Comment:
The method docstring for `APIReleaseClient{}.CreateOrUpdate()` says that `ifMatch` is required when updating a release; should it perhaps be `"*"` like it is for the `Delete()` method, or something else?